performance_manager/lib/rt_vehicle_positions.py

```diff
@@ -275,3 +290,45 @@ def merge_vehicle_position_events(
                     orient="records"
                 ),
             )
+
+
+def process_vehicle_positions(
+    paths_to_load: Dict[str, Dict[str, List]], sql_session: sessionmaker
+) -> None:
+    """
+    process a bunch of vehicle position files
+    create events for them
+    merge those events with existing events
+    """
+    for folder, folder_data in paths_to_load.items():
+        ids = folder_data["ids"]
+        paths = folder_data["paths"]
+
+        try:
+            new_events = get_vp_dataframe(paths)
+            logging.info(
+                "Size of dataframe from %s is %d", folder, new_events.shape[0]
+            )
+            new_events = transform_vp_dtyes(new_events)
+            logging.info(
+                "Size of dataframe with updated dtypes is %d",
+                new_events.shape[0],
+            )
+            new_events = transform_vp_timestamps(new_events)
+            logging.info(
+                "Size of dataframe with transformed timestamps is %d",
+                new_events.shape[0],
+            )
+
+            merge_vehicle_position_events(str(folder), new_events, sql_session)
+        except Exception as e:
+            logging.info("Error Processing Vehicle Positions")
+            logging.exception(e)
+        else:
+            update_md_log = (
+                sa.update(MetadataLog.__table__)
+                .where(MetadataLog.pk_id.in_(ids))
+                .values(processed=1)
+            )
+            with sql_session.begin() as session:  # type: ignore
+                session.execute(update_md_log)
```

Comment:
this does all of the processing. its uses all the above methods and is the only one in lib/__init__.py.